src/Carousel.js

```diff
@@ -112,7 +112,12 @@ class Carousel extends React.Component {
   }
 
   componentDidUpdate(prevProps, prevState) {
-    if (prevState.activeIndex === this.state.activeIndex) return;
+    if (
```

Comment:
better solution as mentioned in the comments

Reply:
Hi,

I can implement the solution you propose indeed but there will be one side effect (should it be considered as a breaking change?): it will restart the interval to 0 at every change. So, let's say you set an interval at 5s and, after 4s, you change the interval to 10s, it will wait 10 more seconds before transitioning. So, in total, it will be 14 seconds before the first transition whereas the current behaviour is different: it does the first transition after 5 seconds and a second transition after 10 seconds.

I guess that if we want to avoid this side effect, we should check that we are in a case of stopping/restarting the carousel. For instance:

```javascript
 componentDidUpdate(prevProps, prevState) {
    if (prevProps.interval !== this.props.interval && (prevProps.interval === false || this.props.interval === false)) {
      this.setInterval();
    }
    if (prevState.activeIndex === this.state.activeIndex) return;
    this.setInterval();
  }
```

Would you be okay with this proposal?

Reply:
`&& (prevProps.interval === false || this.props.interval === false)) {` this will be the case every time we stop and restart right? which means this is always going to be true I guess.

Reply:
> this will be the case every time we stop and restart right?

Yes indeed and I think that's what we want. In that way we fix both issues: the one described in https://github.com/reactstrap/reactstrap/issues/2779 and the one you mentioned in your first comment.

> which means this is always going to be true I guess (as in on the edge case you mentioned).

I don't think so. Or I'm missing something. :thinking:. As far as I understand, `prevProps.interval` would be equal to `5000` and `this.props.interval` to `10000` in my example. So, in that case, `&& (prevProps.interval === false || this.props.interval === false)) {` is going to be false.

Reply:
My bad. Sure, sounds good to me.